File: split/showwhatever.pas
procedure showwhatever;
label
  50;
var
  p: halfword;
begin
  case curchr of
    3:
    begin
      begindiagnostic;
      showactivities;
    end;
    1:{1296:}

    begin
      scaneightbitint;
      begindiagnostic;
      printnl(1253);
      printint(curval);
      printchar(61);
      if eqtb[3678 + curval].hh.rh = 0 then
        print(410)
      else
        showbox(eqtb[3678 + curval].hh.rh);
    end{:1296}
;
    0:{1294:}

    begin
      gettoken;
      if interaction = 3 then;
      printnl(1247);
      if curcs <> 0 then
      begin
        sprintcs(curcs);
        printchar(61);
      end;
      printmeaning;
      goto 50;
    end{:1294}
;
    others:{1297:}

    begin
      p := thetoks;
      if interaction = 3 then;
      printnl(1247);
      tokenshow(29997);
      flushlist(mem[29997].hh.rh);
      goto 50;
    end{:1297}

  end;
  {1298:}
enddiagnostic(true);
  begin
    if interaction = 3 then;
    printnl(262);
    print(1254);
  end;
  if selector = 19 then
    if eqtb[5292].int <= 0 then
    begin
      selector := 17;
      print(1255);
      selector := 19;
    end{:1298}
;
  50:
    if interaction < 3 then
    begin
      helpptr := 0;
      errorcount := errorcount - 1;
    end
    else if eqtb[5292].int > 0 then
    begin
      helpptr := 3;
      helpline[2] := 1242;
      helpline[1] := 1243;
      helpline[0] := 1244;
    end
    else
    begin
      helpptr := 5;
      helpline[4] := 1242;
      helpline[3] := 1243;
      helpline[2] := 1244;
      helpline[1] := 1245;
      helpline[0] := 1246;
    end;
  error;
end;

File: cpp/showwhatever.cpp
#include "showwhatever.h"
#include "begindiagnostic.h"
#include "showactivities.h"
#include "enddiagnostic.h"
#include "printnl.h"
#include "scaneightbitint.h"
#include "print.h"
#include "printint.h"
#include "printchar.h"
#include "showbox.h"
#include "gettoken.h"
#include "printmeaning.h"
#include "sprintcs.h"
#include "thetoks.h"
#include "flushlist.h"
#include "tokenshow.h"
#include "error.h"
#include "texte.h"

void showwhatever(void)
{
	halfword p;
	switch (curchr)
	{
		case 3:
			begindiagnostic();
			showactivities();
			enddiagnostic(true);
			printnl("! ");
			print("OK");
			if (selector == term_and_log && int_par(tracing_online_code) <= 0)
				selector = term_only;
			print(" (see the transcript file)");
			selector = term_and_log;
			break;
		case 1:
			scaneightbitint();
			begindiagnostic();
			printnl("> \\box");
			printint(curval);
			printchar('='); 
			if (box(curval) == 0)
				print("void");
			else
				showbox(box(curval));
			enddiagnostic(true);
			printnl("! ");
			print("OK");
			if (selector == term_and_log && int_par(tracing_online_code) <= 0)
				selector = term_only;
			print(" (see the transcript file)");
			selector = term_and_log;
			break;
		case 0:
			gettoken();
			printnl("> ");
			if (curcs)
			{
				sprintcs(curcs);
				printchar('=');
			}
			printmeaning();
			break;
		default:
			p = thetoks();
			printnl("> ");
			tokenshow(temp_head);
			flushlist(link(temp_head));
			break;
	}
	if (interaction < error_stop_mode)
	{
		helpptr = 0;
		errorcount--;
	}
	else 
		if (int_par(tracing_online_code) > 0)
		{
			helpptr = 3;
			helpline[2] = txt("This isn't an error message; I'm just \\showing something.");
			helpline[1] = txt("Type `I\\show...' to show more (e.g., \\show\\cs,");
			helpline[0] = txt("\\showthe\\count10, \\showbox255, \\showlists).");
		}
		else
		{
		helpptr = 5;
			helpline[4] = txt("This isn't an error message; I'm just \\showing something.");
			helpline[3] = txt("Type `I\\show...' to show more (e.g., \\show\\cs,");
			helpline[2] = txt("\\showthe\\count10, \\showbox255, \\showlists).");
			helpline[1] = txt("And type `I\\tracingonline=1\\show...' to show boxes and");
			helpline[0] = txt("lists on your terminal as well as in the transcript file.");
		}
	error();
}
